pkg/session/session.go

```diff
@@ -117,10 +118,18 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 	sessionMU.Lock()
 	defer sessionMU.Unlock()
 
-	sessionKey := params.server + params.userinfo.Username() + params.datacenter
+	userPassword, _ := params.userinfo.Password()
+	h := sha256.New()
+	h.Write([]byte(userPassword))
+	hashedUserPassword := string(h.Sum(nil))
+	sessionKey := fmt.Sprintf("%s%s%s%s", params.server, params.datacenter, params.userinfo.Username(),
+		hashedUserPassword)
 	if cachedSession, ok := sessionCache.Load(sessionKey); ok {
 		s := cachedSession.(*Session)
-		logger = logger.WithValues("server", params.server, "datacenter", params.datacenter)
+		logger = logger.WithValues(
```

Comment:
Can we add those values to l.117? Then we don't have to do it here and in newClient and newManager

Reply:
@sbueringer Make sense, this looks more neat, didn't notice it before, will update in next patch